### ISSUE TITLE
LightTypeTag behaviour doesn't follow compiler subtyping rules in case of literal and/or identifier types  

### ISSUE THREAD

**Issue by ak0rz** [zio/izumi-reflect#284]
Related to zio/zio#6711

scala 2.13 ([scastie](https://scastie.scala-lang.org/RXFOqT2xSx24Fx2VeKF9yw))

```scala
import izumi.reflect._

val a: "a" = "a"
type a = a.type

locally {
  // here shown actual scala compiler behaviour
  // if it compiles, it shows that theese properties are actually correct
  val _ = implicitly["a" <:< a]
  val _ = implicitly[a <:< "a"]
  val _ = implicitly[a =:= "a"]
  val _ = implicitly[a =:= a]
  val _ = implicitly["a" =:= "a"]
  val _ = implicitly["a" <:< String]
  val _ = implicitly[a <:< String]
  // val _ = implicitly[String <:< a] doesn't compile
  // val _ = implicitly[String <:< "a"] doesn't compile
}

// here we try to check the same properties in runtime over
// LightTypeTag (expands to izumi.reflect.macrortti.LightTypeTag)
println((Tag["a"].tag <:< Tag[a].tag).toString + " // expected to be true")
println((Tag[a].tag <:< Tag["a"].tag).toString + " // expected to be true") 
println((Tag[a].tag =:= Tag["a"].tag).toString + " // expected to be true")
println((Tag["a"].tag =:= Tag["a"].tag).toString + " // expected to be true")
println((Tag[a].tag =:= Tag[a].tag).toString + " // expected to be true")
println((Tag["a"].tag <:< Tag[String].tag).toString + " // expected to be true")
println((Tag[a].tag <:< Tag[String].tag).toString + " // expected to be true")
println((Tag[String].tag <:< Tag["a"].tag).toString + " // expected to be false")
println((Tag[String].tag <:< Tag[a].tag).toString + " // expected to be false")

println()
println(Tag["a"].tag.debug())
println(Tag[a].tag.debug())
```

output

```
false // expected to be true
false // expected to be true
false // expected to be true
true // expected to be true
true // expected to be true
true // expected to be true
true // expected to be true
false // expected to be false
false // expected to be false

⚙️ begin : "a"
⚡️bases:
 - "a" ->   
   * java.io.Serializable
   * java.lang.CharSequence
   * java.lang.String
   * java.lang.constant.Constable
   * java.lang.constant.ConstantDesc
   * java.lang.Comparable[=java.lang.String]
⚡️inheritance:
 - "a" ->   
   * java.io.Serializable
   * java.lang.CharSequence
   * java.lang.Comparable
   * java.lang.String
   * java.lang.constant.Constable
   * java.lang.constant.ConstantDesc
⚙️ end 
⚙️ begin : Playground::a
⚡️bases:
 - Playground::a ->   
   * java.io.Serializable
   * java.lang.CharSequence
   * java.lang.String
   * java.lang.constant.Constable
   * java.lang.constant.ConstantDesc
   * java.lang.Comparable[=java.lang.String]
⚡️inheritance:
 - Playground::a ->   
   * java.io.Serializable
   * java.lang.CharSequence
   * java.lang.Comparable
   * java.lang.String
   * java.lang.constant.Constable
   * java.lang.constant.ConstantDesc
⚙️ end 
```

**Comment by pshirshov** [comment]
So, this is fixed for 2.13, still broken on Dotty though.

**Comment by pshirshov** [comment]
Okay, I fixed this for dotty, the fix may be imperfect but it works.
